### PR TITLE
Add project configuration

### PR DESCRIPTION
**Pull request by ant31**
1. It's easier to have a file to manage server-side configuration (configmap) instead set 20+ env vars

### CODE REVIEW

**Comment by ant31** [comment]
foo

**Comment by ant31** [comment]
@samba 

The PR enable the configuration via a config file instead of env variable.

It includes some code to prepare follow-up pr for: 
- fetch per-repository configuration
- trigger or re-trigger builds from github via labels / comments
- rules to who can trigger builds (e.g contributors only)

File: deploy/failfast/resources/ff-api.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        - name: CELERY_BROKER
+          value: redis://failfast-redis.hooktest.svc.cluster.local:6379
+        - name: FFCI_CONF_FILE
+          value: /etc/failfast-ci/failfast-ci.yaml
+        - name: GITHUB_INTEGRATION_PEM
+          valueFrom:
+            secretKeyRef:
+              key: integration_pem
+              name: failfast-ci
+        image: quay.io/failfast-ci/failfast-cibuild:pr-36-failfast-configuration-8b7ba189
```
Comment (samba) [inline deploy/failfast/resources/ff-api.yaml]: The image label here seems like something we'd want to generalize for use in other deployments. Stable? (TODO?)

**Comment by ant31** [comment]
@samba I've removed the default integration_id. 
About the deployment, it's using `kubectl kinflate` and added small documentation on how to use it:
https://github.com/failfast-ci/failfast-api/blob/36e2ee5ad9b3f93d0f2854d0a483da91cf9169a5/deploy/failfast/Readme.md

**Pull request merged**